Service/RateLimitInfo.php

```diff
@@ -55,4 +65,44 @@ public function setResetTimestamp($resetTimestamp)
     {
         $this->resetTimestamp = $resetTimestamp;
     }
+
+    /**
+     * Return true if the action is blocked
+     *
+     * @return bool
+     */
+    public function isBlocked()
+    {
+        return $this->blocked;
+    }
+
+    /**
+     * Set block the action
+     *
+     * @param bool $blocked
+     */
+    public function setBlocked($blocked)
+    {
+        $this->blocked = (bool)$blocked;
+    }
+
+    /**
+     * Return the key of rate limit
+     *
+     * @return string
+     */
+    public function getKey()
+    {
+        return $this->key;
+    }
+
+    /**
+     * Set the key into rate limit
+     *
+     * @param string $key
+     */
+    public function setKey($key)
+    {
+        $this->key = (string)$key;
```

Comment:
I'm not sure why you want to set the key?

Reply:
We need the key there.

```php
    # Service/Storage/StorageInterface.php

    /**
     * Set block for the call
     *
     * @param RateLimitInfo $rateLimitInfo
     * @param integer       $periodBlock
     *
     * @return bool
     */
    public function setBlock(RateLimitInfo $rateLimitInfo, $periodBlock);
```

If we passed only the `key` and `periodBlock` into `setBlock` method we would have to load cache data before setting block in some storages. It's not a good idea to load data if we have already them.

Reply:
I think it would be better for this to work like the other functions in RateLimitService and have $key as the first parameter